### PR TITLE
[registrar] Make sure to release the return value from xamarin_get_parameter_type.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Before:

    There were 258046 MonoObjects created, 235142 MonoObjects freed, so 22904 were not freed. (dynamic registrar)
    There were 205804 MonoObjects created, 204193 MonoObjects freed, so 1611 were not freed. (static registrar)

After:

    There were 258054 MonoObjects created, 235172 MonoObjects freed, so 22882 were not freed. (dynamic registrar)
    There were 205804 MonoObjects created, 205190 MonoObjects freed, so 614 were not freed. (static registrar)

### CODE REVIEW

File: runtime/runtime.m [hunk 1]
```diff
@@ -270,8 +270,10 @@
 	if (index == -1) {
 		p = mono_signature_get_return_type (msig);
 	} else {
-		for (int i = 0; i < index + 1; i++)
+		for (int i = 0; i < index + 1; i++) {
+			xamarin_mono_object_release (&p);
 			p = mono_signature_get_params (msig, &iter);
+		}
```
Comment (spouliot) [inline runtime/runtime.m:276]: that seems duplicated inside https://github.com/xamarin/xamarin-macios/pull/11730

same comments
* first iteration p == NULL, is that fine ?
* last iteration, should not `p` be released ?

Comment (spouliot) [reply runtime/runtime.m:276]: both answered in https://github.com/xamarin/xamarin-macios/pull/11730

Comment (rolfbjarne) [reply runtime/runtime.m:276]: yeah, I repeated myself...

Hopefully there won't be any merge conflicts!

**Comment by rolfbjarne** [comment]
Test failures are unrelated:

* monotouch-test/Mac [dotnet]/Debug (CoreCLR) [dotnet]: will be fixed with #11732 and #11733.
* monotouch-test/Mac [dotnet]/Debug (CoreCLR, static registrar) [dotnet]: will be fixed with #11732 and #11733.

**Pull request merged**